src/core/annotation.js

```diff
@@ -224,11 +236,7 @@ var Annotation = (function AnnotationClosure() {
      * @see {@link shared/util.js}
      */
     setFlags: function Annotation_setFlags(flags) {
-      if (isInt(flags)) {
-        this.flags = flags;
-      } else {
-        this.flags = 0;
-      }
+      this.flags = (isInt(flags) && flags > 0) ? flags : 0;
```

Comment:
I suppose so, together with the slight change of the `_hasFlag` method in the new patch.
I should perhaps not have sneaked this into the current patch, but given that I needed to touch most of those methods anyway, it seemed better to reduce code churn a bit.